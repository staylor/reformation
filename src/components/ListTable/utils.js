// eslint-disable-next-line
export const formatDate = date => {
  const d = new Date(date);
  const month = d.getMonth() + 1;
  const day = d.getDate();
  const min = d.getMinutes();
  const hour = d.getHours();
  return `${month < 10 ? `0${month}` : month}/${day < 10 ? `0${day}` : day}/${d.getFullYear()}
  ${' '}at${' '}
  ${hour % 12}:${min < 10 ? `0${min}` : min}${hour < 12 ? 'am' : 'pm'}`;
};
