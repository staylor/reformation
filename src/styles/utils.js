import { styled, css } from 'pretty-lights';
import facepaint from 'facepaint';
import themeUtils from 'styles/theme';
import responsive from 'styles/responsive';

export const baseHeadingClass = css`
  display: block;
  font-weight: ${themeUtils.fonts.weight.bold};
  letter-spacing: 0.3px;
  line-height: 1.3;
  margin-bottom: 10px;

  ${responsive.desktop} {
    margin-bottom: ${themeUtils.padding}px;
  }
`;

export const h1Class = css`
  ${baseHeadingClass};
  font-size: 30px;

  ${responsive.desktop} {
    font-size: 36px;
  }
`;

export const h2Class = css`
  ${baseHeadingClass};
  font-family: ${themeUtils.fonts.futura};
  font-size: 24px;
`;

export const h3Class = css`
  ${baseHeadingClass};
  font-family: ${themeUtils.fonts.futura};
  font-size: 20px;
`;

export const h4Class = css`
  ${baseHeadingClass};
  font-family: ${themeUtils.fonts.futura};
  font-size: 18px;
`;

export const Heading = styled.h2`
  ${baseHeadingClass};
  color: ${themeUtils.colors.black};
  font-family: ${themeUtils.fonts.futura};
  font-size: 25px;
`;

export const LoadMore = styled.button`
  appearance: none;
  background: ${themeUtils.colors.white};
  border: 1px solid ${themeUtils.colors.detail};
  box-sizing: border-box;
  color: ${themeUtils.colors.inactive};
  cursor: pointer;
  font-size: 16px;
  height: 32px;
  line-height: 1;
  text-align: center;
  text-transform: uppercase;
  transition: 400ms;
  width: 80px;

  &:hover,
  &:active,
  &:focus {
    border: 1px solid ${themeUtils.colors.black};
    color: ${themeUtils.colors.black};
    outline: 0 none;
  }
`;

export const baseButtonClass = css`
  appearance: none;
  border-radius: 3px;
  box-sizing: border-box;
  cursor: pointer;
  display: inline-block;
  font-size: 13px;
  height: 28px;
  line-height: 26px;
  margin: 0;
  outline: 0;
  padding: 0 10px 1px;
  text-decoration: none;
  transition-duration: 0.05s;
  transition-property: border, background, color;
  transition-timing-function: ease-in-out;
  vertical-align: top;
  white-space: nowrap;

  &:active {
    transform: translateY(1px);
  }
`;

const pseudo = facepaint([':active', ':focus', ':hover']);

const generateButtonColors = c =>
  css(
    pseudo({
      background: [c.background, c.active.background, c.focus.background, c.hover.background],
      borderColor: [c.border, c.active.border, c.focus.border, c.hover.border],
      boxShadow: [c.boxShadow, c.active.boxShadow, c.focus.boxShadow],
      color: [c.color, null, c.focus.color, c.hover.color],
    })
  );

export const buttonColors = generateButtonColors(themeUtils.buttons.base);

export const Button = styled.button`
  ${baseButtonClass};
  ${buttonColors};
`;

export const largeButtonClass = css`
  ${baseButtonClass};
  height: 30px;
  line-height: 28px;
  padding: 0 12px 2px;
`;

const primaryButtonColors = generateButtonColors(themeUtils.buttons.primary);

export const PrimaryButton = styled.button`
  ${largeButtonClass};
  ${primaryButtonColors};
  text-shadow: ${themeUtils.buttons.primary.textShadow};
`;

export const smallButtonClass = css`
  ${baseButtonClass};
  font-size: 11px;
  height: 24px;
  line-height: 22px;
  padding: 0 8px 1px;
`;

export const SecondaryButton = styled.button`
  ${smallButtonClass};
  ${buttonColors};
`;
