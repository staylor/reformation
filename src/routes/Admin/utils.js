import { useState } from 'react';
import { useLocation, useParams, useHistory } from 'react-router-dom';
import { useQuery, useMutation } from '@apollo/client';
import { parse, stringify } from 'query-string';
import { offsetToCursor } from 'utils/connection';

export const useQueryParams = (keys = []) => {
  const location = useLocation();
  const queryParams = parse(location.search);
  const variables = {};
  keys.forEach(key => {
    if (queryParams[key]) {
      if (key === 'year') {
        variables[key] = parseInt(queryParams[key], 10);
      } else {
        variables[key] = queryParams[key];
      }
    }
  });
  return variables;
};

export const usePageOffset = variables => {
  const params = useParams();
  if (params.page) {
    const pageOffset = parseInt(params.page, 10) - 1;
    if (pageOffset > 0) {
      variables.after = offsetToCursor(pageOffset * variables.first - 1);
    }
  }
  return variables;
};

export const useAdminQuery = (query, variables) =>
  useQuery(query, {
    variables,
    // This ensures that the tables are up to date when items are mutated.
    // The alternative is to specify refetchQueries on all mutations.
    fetchPolicy: 'cache-and-network',
  });

export const usePropUpdate = ({ prop, pathname }) => {
  const history = useHistory();
  return value => {
    const p = {};
    if (value) {
      p[prop] = value;
    }
    history.push({
      pathname,
      search: stringify(p),
    });
  };
};

export const useEditQuery = (query, variables = {}) => {
  const params = useParams();
  return useAdminQuery(query, { ...variables, id: params.id });
};

export const useSubmitEdit = ({ mutation }) => {
  const [mutate] = useMutation(mutation);
  const [message, setMessage] = useState(null);

  const onSubmit = data => (e, updates) => {
    e.preventDefault();

    mutate({
      variables: {
        id: data.id,
        input: updates,
      },
    })
      .then(() => {
        setMessage('updated');
        document.documentElement.scrollTop = 0;
      })
      .catch(() => setMessage('error'));
  };

  return { onSubmit, message };
};

export const useSubmitNew = ({ mutation, path, key }) => {
  const [mutate] = useMutation(mutation);
  const [message, setMessage] = useState(null);
  const history = useHistory();

  const onSubmit = (e, updates) => {
    e.preventDefault();

    mutate({
      variables: {
        input: updates,
      },
    })
      .then(({ data }) => {
        history.push({
          pathname: `/${path}/${data[key].id}`,
        });
      })
      .catch(() => setMessage('error'));
  };

  return { message, onSubmit };
};

export const useSubmitDelete = ({ mutation, query: { refetch } }) => {
  const [mutate] = useMutation(mutation);

  return ids => e => {
    e.preventDefault();

    mutate({
      variables: {
        ids,
      },
    }).then(() => {
      refetch();
    });
  };
};
