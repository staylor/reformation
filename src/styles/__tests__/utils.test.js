import React from 'react';
import { mount } from 'enzyme';
import { cx } from 'pretty-lights';
import { headingStyles, Heading, largeButtonStyles, smallButtonStyles } from '../utils';

describe('Style utils', () => {
  test('headingStyles', () => {
    const wrapper = mount(<h1 className={headingStyles}>Hello</h1>);

    expect(wrapper).toMatchSnapshot();
  });

  test('Heading', () => {
    const wrapper = mount(<Heading>Hello</Heading>);

    expect(wrapper).toMatchSnapshot();
  });

  test('cx', () => {
    const wrapper = mount(
      <a
        href="http://foo.com"
        className={cx({
          [largeButtonStyles]: typeof largeButtonStyles === 'string',
          [smallButtonStyles]: typeof smallButtonStyles !== 'string',
        })}
      >
        Hello
      </a>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
