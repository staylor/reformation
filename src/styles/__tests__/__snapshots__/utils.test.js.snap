// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Style utils Heading 1`] = `
.lights-0 {
  display: block;
  font-weight: 700;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 1.3;
  margin-bottom: 10px;
  color: #000;
  font-family: 'futura-pt','Helvetica Neue',sans-serif;
  font-size: 25px;
}

@media only screen and (min-width:1024px) {
  .lights-0 {
    margin-bottom: 15px;
  }
}

<Heading>
  <h2
    className="lights-0 lights-1"
  >
    Hello
  </h2>
</Heading>
`;

exports[`Style utils cx 1`] = `
.lights-0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border-radius: 3px;
  box-sizing: border-box;
  cursor: pointer;
  display: inline-block;
  font-size: 13px;
  height: 28px;
  line-height: 26px;
  margin: 0;
  outline: 0;
  padding: 0 10px 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition-duration: 0.05s;
  transition-duration: 0.05s;
  -webkit-transition-property: border,background,color;
  transition-property: border,background,color;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  vertical-align: top;
  white-space: nowrap;
  height: 30px;
  line-height: 28px;
  padding: 0 12px 2px;
}

.lights-0:active {
  -webkit-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
}

<a
  className="lights-0"
  href="http://foo.com"
>
  Hello
</a>
`;

exports[`Style utils headingStyles 1`] = `
.lights-0 {
  display: block;
  font-weight: 700;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 1.3;
  margin-bottom: 10px;
}

@media only screen and (min-width:1024px) {
  .lights-0 {
    margin-bottom: 15px;
  }
}

<h1
  className="lights-0"
>
  Hello
</h1>
`;
